app.py

```diff
@@ -232,57 +233,81 @@ def find_closest_match(ideal_features, playlist_features):
 
     return ideal_id  # some track id
 
+def gather_song_set(playlist_id, ideal_audio_features, spotipy_manager = None):
+    """ Obtains a list of songs depending what playlist is selected """
+    local_spotipy = get_spotipy() if spotipy_manager is None else spotipy_manager
 
-# Method that starts play
-def queue_from_playlist(ideal_audio_features, data):
-    """queues a song based on the ideal audio features"""
-    playlist_id = data["playlistID"]
+    if playlist_id in ["discover_mode", 1, "1"]: # OR WAHATEVER WE WANTED
+        songs = discover_song(ideal_audio_features, local_spotipy)
 
-    if playlist_id in ["discover_mode", 1, "liked_songs", "1"]:
-        # In the case this is a new playlist, we don't need to recommend anything
-        # (as of right now). Just transfer playback
-        discover_song(ideal_audio_features)
-        return
+    elif playlist_id in ["discover_mode", 1, "liked_songs", "1"]:
+        songs = local_spotipy.current_user_saved_tracks(limit=100)
+
+    else:
+        tracks = local_spotipy.playlist(playlist_id)
+        songs = tracks["tracks"]["items"]
+
+    return songs
+
+
+
+def filter_songs(songs, ideal_audio_features, spotipy_manager = None):
+    """ filters down the possible set of songs to the ideal one, return best match """
+    local_spotipy = get_spotipy() if spotipy_manager is None else spotipy_manager
 
-    local_spotipy = get_spotipy()
     audio_features = None
 
-    if local_spotipy is not None and (
-        local_spotipy.current_playback() is None
-        or local_spotipy.current_playback()["progress_ms"] >= UPDATE_SONG_TIME_MS
-    ):
-        # get the songs from the playlist
-        # tracks = localSP.playlist(playlistID, fields="tracks,next")
-        tracks = local_spotipy.playlist(playlist_id)
-        # print('Custom playlist!')
 
-        # print(tracks)
-        track_ids = []
+    # get the songs from the playlist
+    # tracks = local_spotipy.playlist(playlist_id)
 
-        for (song, i) in zip(tracks["tracks"]["items"], range(100)):
-            if i >= 0 and song is not None:
-                track_ids.append(song["track"]["uri"])
+    track_ids = []
 
-        print(track_ids)
-        # get the audio features
-        audio_features = local_spotipy.audio_features(tracks=track_ids)
+    for (song, i) in zip(songs, range(100)):
```

Comment:
You can use enumerate to return an iterable here instead of zipping a whole new one